kitsune/coolsearch/static/coolsearch/js/components/SearchForm.jsx

```diff
@@ -0,0 +1,86 @@
+/*
+React component that controls the behavior of the Search form.
+Handles the tabs of the form and the triggering of search queries.
+*/
+
+import SwipeViews from 'react-swipe-views';
+
+import SearchFormWiki from './SearchFormWiki.jsx';
+import SearchFormQuestion from './SearchFormQuestion.jsx';
+import SearchFormForum from './SearchFormForum.jsx';
+
+import QueryStore from '../stores/QueryStore.js';
+
+import QueryActionCreator from '../actions/QueryActionCreator.js';
+import SearchActionCreator from '../actions/SearchActionCreator.js';
+
+
+var getStateFromStores = function () {
+  return {
+    query: QueryStore.getCurrentQuery()
+  };
+};
+
+
+export default class SearchForm extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = getStateFromStores();
+    this.currentForm = 'wiki';
+
+    this.startQuery = _.debounce(this.startQuery.bind(this), 300);
+  }
+
+  componentDidMount() {
+    QueryStore.addChangeListener(this._onChange.bind(this));
+  }
+
+  _onChange() {
+    this.setState(getStateFromStores());
+    this.startQuery();
+  }
+
+  getResults(e) {
+    if (e) {
+      e.preventDefault();
+    }
+
+    SearchActionCreator.runSearch(this.currentForm, this.state.query);
+  }
+
+  /** A debounced proxy to getResults(). */
```

Comment:
This comment isn't strictly accurate, since the debouncing happens elsewhere. In fact, I think that this method definition isn't needed. The constructor could have the same effect with

```js
this.startQuery = _.debounce(this.getResults.bind(this), 300);
```